instruqt/delivering-as-an-appliance/02-customize-for-your-app/assignment.md

```diff
@@ -16,6 +20,184 @@ timelimit: 300
+available, degraded one but not all replicas are ready, and unavailable if
+none are ready. If there are replicas are from different versions then the
+resources is updating.</td></tr>
+<tr><td><code>DaemonSet</code></td><td>Similar to </code>Deployment</code> or </code>StatfulSet</code> but based on
+the daemon pods</td></tr>
+<tr><td><code>Service</code><br/><code>Ingress</code></td><td>Services are ready if all endpoints are ready and any
+needed load balancers have been assigned. They are degraded if at least one
+endpoint is ready when more are requests, and unavailable if none are ready
+and no load balancer(s) have been assigned. Ingress is based on the same
+criteria for it's backing service.</td></tr>
+<tr><td><code>PersistentVolmeClaim</code></td><td>A PVC is ready if the claim is bound and
+unavailable if it is either lost or pending. Those are the only two states.</td></tr>
+</tbody>
+<table>
+
+Slackernews has three components in it's architecture: the application, and
+NGINX web server, and a Postgres database. It can optionally have an ingress
+as well, but will not in the virtual appliance. Let's add status informers for
+the application and the web server.
+
+
+In the "Release Editor" tab, add the following to the `spec` section of the
+manifest.
+
+```
+  statusInformers:
+    - deployment/slacknerews
+    - deployment/slackernews-nginx
+```
+
+![Adding the application metadata manifest](../assets/adding-status-informers.png)
+
+We haven't provided a status informer For Postgres since the customer can
+choose to bring their own database. We'll add it in when we work with the
+customer configuration in the next step of this lab.
+
+Releasing Your Application
+==========================
+
+We're still not going to install our appliance, but let's release these
+changes to get into the habit or releasing changes quickly.
+
+First we want to bump the version in our Helm chart. You could consider this
+part optional since we didn't update the chart. I recommend you do it, since
+different customer may install the application in different ways. It will be
+much easier for your team if Helm installs using version `0.6.1` and appliance
+installs using version `0.6.1` are using the same bits.
+
+You can bump the chart version in your editor, or just run the following two
+commands
+
+```
+yq -i '.version = "0.6.1"' slackernews/Chart.yaml
+yq -i '.spec.chart.chartVersion = "0.6.1"' release/slackernews-chart.yaml
+```
+
+You need to edit both files to assure that the appliance refers to the current
+chart. You also need to repackage you're Helm chart and make sure your old
+package isn't still in the release directory.
+
+```
+helm package -u slackernews -d release
+rm release/slackernews-0.6.0.tgz
+```
+
+Create a new release, and promote it directly to the "Unstable" channel.
+
+```
+replicated release create --promote Unstable --chart ${HOME_DIR}/release/slackernews-0.6.1.tgz --version 0.6.1 \
```

Comment:
Isn't this a Helm-only release?

Reply:
This whole command doesn't work for me, and I can't proceed. Looks like `HOME_DIR` is empty, so it tries to find the chart at `/release/slackernews-0.6.1.tgz`. I don't know how that variable worked in the previous step though.
```
replicant@shell:~$ helm package -u slackernews -d release                                                                                                                        
Pulled: registry.replicated.com/library/replicated:1.0.0-beta.26
Digest: sha256:221b4d6ebb329e5d7e2d0bed4dafd59a130c77a58b8cefa24e6161e035ebb53d
Successfully packaged chart and saved it to: release/slackernews-0.6.1.tgz
replicant@shell:~$ replicated release create --promote Unstable --chart ${HOME_DIR}/release/slackernews-0.6.1.tgz --version 0.6.1 \
  --release-notes "Configures appliance branding, status, and dashboard links" \
  --app ${REPLICATED_APP} --token ${REPLICATED_API_TOKEN}
You are creating a release that will only be installable with the helm CLI.
For more information, see 
https://docs.replicated.com/vendor/helm-install#about-helm-installations-with-replicated

  _ Reading chart from /release/slackernews-0.6.1.tgz _  
ERROR: open /release/slackernews-0.6.1.tgz: no such file or directory
```

Reply:
Copy/paste error on my part. I copied it out of the script that I can test for the lab and didn't fix that up. You can set $HOME_DIR to $HOME to workaround but I'll fix.